pkg/services/searchV2/index.go

```diff
@@ -929,9 +929,16 @@ func (l sqlDashboardLoader) LoadDashboards(ctx context.Context, orgID int64, das
 	defer cancelLoadingDashboardCtx()
 
 	dashboardsChannel := l.loadAllDashboards(loadingDashboardCtx, limit, orgID, dashboardUID)
+	defer close(dashboardsChannel)
```

Comment:
probably not worth changing, but i'm wondering if `loadAllDashboards` should close the channel when it's finished writing instead of closing it in the reader?

it seems like then `res.end` could be replaced with checking `done` in the select: `case res, done = <-dashboardsChannel:`

Reply:
thats way cleaner, done! I removed the context cancellation check in the main loop, I don't think its necessary anymore.